cdap-docs/developers-manual/source/building-blocks/workflows.rst

```diff
@@ -86,12 +94,114 @@ implement the ``run()`` method::
 The custom action then can be added to the workflow using the ``addAction()`` method as
 shown above.
 
+.. _workflow-concurrent:
+
 Concurrent Workflows
 --------------------
 By default, a workflow runs sequentially. Multiple instances of a workflow can be run
 concurrently. To enable concurrent runs for a workflow, set its runtime argument
 ``concurrent.runs.enabled`` to ``true``.
 
+
+.. _workflow_token:
+
+Workflow Tokens
+===============
+
+In addition to passing the control flow from one node to the next, a **workflow token** is
+passed, available to each of the programs in the workflow. This allows programs to:
+
+- pass custom data (such as metrics, a status, or an error code) from one program in the 
+  workflow to the next subsequent programs; 
+- query and set the data in the token;
+- fetch the data from the token which was set by a specific node; and
+- determine the name of the node which most recently set the token value for a specific key;
+  e.g., the node who last set the ERROR flag in the token.
+  
+The last example is intended to allow appropriate action to be taken in response, such as
+logging, modifying the conditional execution, or terminating the execution of workflow.
+
+Execution in the workflow can be made conditional, based on the information contained in
+the token. Execution can be terminated if a node in the workflow produces unexpected
```

Comment:
I think its better to remove the line `Execution can be terminated if a node in the workflow produces unexpected..`,  as we don't have that capability yet.